docs/design/satellite-svc-separation.md

```diff
@@ -0,0 +1,93 @@
+# Satellite Service Separation
+
+## Abstract
+
+The goal of the design doc is to describe the work necessary to make the Satellite horizontally scalable.
+
+## Background
+
+Currently all Satellite services run in a single binary in a single container. While this is great for development, this is bad for scaling.
+
+Currently the Satellite is a single binary made of up the following services:
+- version
+- overlay
+- metainfo
+- inspector
+- orders
+- repair (includes checker and repairer)
+- audit
+- garbage collection (GC)
+- accounting (includes tally and rollup service and the live accounting cache)
+- mail
+- console
+- nodestats
+- note: marketing service is currently on pause and is not included in this doc
+- note: kademlia, discovery, and vouchers are being removed and not included in this doc
+
+The Satellite has the following public GRPC endpoints:
+- metainfo
+- nodestats
+- orders
```

Comment:
- nodes / overlay which nodes use to do the first connection

Reply:
It's part of kademlia at the moment, https://github.com/storj/storj/blob/master/satellite/peer.go#L341... but we'll need something to handle it.